fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -374,6 +374,26 @@ trait Parser[+T, ElemType, Repr] extends ParserResults[T, ElemType] with Precede
   }
 
   /**
+   * Extractor for pattern matching
+   *
+   *  For example:
+   *
+   *  {{{
+   *  val p1 = CharIn('a'to'z').! ~ CharIn('0'to'9').rep(1).!.map(_.toInt)
+   *  List("a42x") match {
+   *    case p1(x: String, y: Int) :: _ => // x is "a", y is 42
+   *  }
+   * }}}
+   */
+  def unapply[Repr <% IndexedSeq[ElemType]](input: Repr): Option[T] = {
```

Comment:
I've just compiled code like this: 
```scala
 def unapply(input: Repr)(implicit formatter: ElemTypeFormatter[ElemType],
                                    converter: ResultConverter[ElemType, Repr]): Option[T] = {
    parse(input) match {
      case Parsed.Success(r, _) => Some(r)
      case _ => None
    }
```
And it seems everything is okay, could you explain the problem with implicits in unapply more detailed?